kotlinx-coroutines-debug/build.gradle

```diff
@@ -39,19 +31,27 @@ java {
 }
 
 jar {
+    setEnabled(false)
     manifest {
         attributes "Premain-Class": "kotlinx.coroutines.debug.AgentPremain"
         attributes "Can-Redefine-Classes": "true"
     }
 }
```

Comment:
I find this very counterintuitive: if a task is not enabled, why does its configuration contribute to anything? I propose to move the `manifest` block to `shadowJar`. If I understood correctly (and from a quick check of the resulting `META-INF/MANIFEST.MF` in the published JAR), this shouldn't change the result, but it would look much more clear.

Reply:
That's what I was proposing to do, but then I noticed the `Class-Path` line disappearing from the manifest.

Reply:
The previous version also does not generate `Class-Path` attribute. 
This is a nice catch, though I cannot convince myself that it actually ever affected anything.

Reply:
This is what I observed as well, yes, just wasn't sure.

Could you please try using the debug module in a minimal project as a dependency and check that everything still works, just to be safe? We do have integration tests for using it as a Java agent, but not for just using `DebugProbes`, right?

Reply:
Restored it back, created https://github.com/Kotlin/kotlinx.coroutines/issues/3361

Will push integration test soon as well